src/Entity/Document.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiFilter;
+use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ApiResource(
+ *     routePrefix="/v3",
+ *     attributes={
+ *         "order": {"createdAt": "DESC"},
+ *         "normalization_context": {
+ *             "groups": {"document_read"}
+ *         },
+ *         "security": "is_granted('ROLE_OAUTH_SCOPE_JEMENGAGE_ADMIN') and is_granted('IS_FEATURE_GRANTED', 'documents')",
+ *     },
+ *     collectionOperations={
+ *         "get": {
+ *             "path": "/documents",
+ *             "maximum_items_per_page": 1000
+ *         }
+ *     },
+ *     itemOperations={
+ *         "get": {
+ *             "path": "/documents/{uuid}",
+ *             "requirements": {"uuid": "%pattern_uuid%"},
+ *             "security": "is_granted('ROLE_OAUTH_SCOPE_JEMENGAGE_ADMIN') and is_granted('IS_FEATURE_GRANTED', 'documents')",
+ *         },
+ *         "get_file": {
+ *             "path": "/documents/{uuid}/file",
+ *             "method": "GET",
+ *             "controller": "App\Controller\Api\DocumentDownloadFileController",
+ *             "requirements": {"uuid": "%pattern_uuid%"},
+ *             "security": "is_granted('ROLE_OAUTH_SCOPE_JEMENGAGE_ADMIN') and is_granted('IS_FEATURE_GRANTED', 'documents')",
+ *         }
+ *     }
+ * )
+ *
+ * @ApiFilter(SearchFilter::class, properties={
+ *     "title": "partial",
+ *     "visibility": "exact",
+ * })
+ *
+ * @ORM\Entity(repositoryClass="App\Repository\DocumentRepository");
+ * @ORM\Table(name="document")
```

Comment:
we can avoid `name` attribut

```suggestion
 * @ORM\Table()
```